compiler/rustc_hir_analysis/src/collect/predicates_of.rs

```diff
@@ -62,6 +62,143 @@ pub(super) fn predicates_of(tcx: TyCtxt<'_>, def_id: DefId) -> ty::GenericPredic
 fn gather_explicit_predicates_of(tcx: TyCtxt<'_>, def_id: LocalDefId) -> ty::GenericPredicates<'_> {
     use rustc_hir::*;
 
+    match tcx.opt_rpitit_info(def_id.to_def_id()) {
+        Some(ImplTraitInTraitData::Trait { fn_def_id, opaque_def_id }) => {
+            let mut predicates = Vec::new();
+            // FIXME we need a query and we need bidir_outlives metadata
+            if opaque_def_id.is_local() {
+                let icx = ItemCtxt::new(tcx, def_id);
+                let hir_id = tcx.hir().local_def_id_to_hir_id(opaque_def_id.expect_local());
+                let node = tcx.hir().get(hir_id);
+                let Node::Item(&Item { kind: ItemKind::OpaqueTy(OpaqueTy { bidir_outlives, .. }), .. }) = node else {
+                    bug!("unexpected {node:?}")
+                };
+
+                for (arg, dup_def) in bidir_outlives {
+                    let orig_region = icx.astconv().ast_region_to_region(arg, None);
```

Comment:
@compiler-errors you told me that this could be done differently. I was reusing this idea https://github.com/rust-lang/rust/pull/112682/files#diff-6d9f47a8bce6ee31fd3e867514607c8503a11cf91616ce418393ba9304a22c16R435. In case Github doesn't take you there line 435 on the right side.

Reply:
I guess you meant `named_bound_var`.

Reply:
I guess that's fine. I think using `named_bound_var` might be easier but I guess it doesn't matter.